pages/_app.jsx

```diff
@@ -9,8 +9,12 @@ function NeighborExpress({ children }) {
   let { state, dispatch } = useContext(CMSContext);
 
   useEffect(() => {
+    import('../cms-cache.json').then((json) => json.records).then((records) => {
+      dispatch({ type: 'set-cached-records', payload: processRecords(records, state) })
+    });
+
```

Comment:
Did you consider making this the initialState in the context directly vs setting it when the app renders?

Reply:
That sounds even faster! I'll look into it

Reply:
Hm, okay looking at this, I wasn't able to get it to be as fast as initialState, since the load from the file is also async -- it was populating the initial state with no records, and the local data wasn't coming any faster than the server data...

I also like having the separate action `set-cached-records` so we make sure we're never overwriting fresh content with stale/cached content.

Reply:
Yup, after double checking that what I suggested made sense (it doesn't), I agree with your implementation here!

Reading:
https://gist.github.com/astoilkov/013c513e33fe95fa8846348038d8fe42
https://medium.com/@albertogasparin/the-pitfalls-of-async-operations-via-react-context-api-ab987d4290e6